File: benchmarks/closure_calls.go
package main

import (
	"fmt"
	"time"
)

func main() {
	t0 := time.Now()
	f := func(x int) int { return x % 7 }
	total := 0
	for i := 0; i < 5000000; i++ {
		total += f(i)
	}
	fmt.Println(total)
	fmt.Printf("bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/closure_calls.rs
use std::time::Instant;

fn main() {
    let t0 = Instant::now();
    let f = |x: i32| x % 7;
    let mut total = 0;
    for i in 0..5000000 {
        total += f(i);
    }
    println!("{}", total);
    println!("bench_ns={}", t0.elapsed().as_nanos());
}
